tests/fixtures/site.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+
+
+@pytest.fixture
+def site_matrix(factories):
+    projects = factories['project'](batch=3)
+    locales = factories['locale'](batch=3)
+    pl_kwa = []
+    resource_kwa = []
+    tr_kwa = []
+    for project in projects:
+        resources = []
+        for i in range(0, 7):
+            resources.append(
+                dict(project=project,
+                     total_strings=(i + 5) ** 3))
+        for locale in locales:
+            pl_kwa.append(dict(project=project, locale=locale))
+            for i, resource in enumerate(resources):
+                tr_kwa.append(
+                    dict(locale=locale,
+                         approved_strings=(i + 2) ** 3,
+                         translated_strings=(i + 1) ** 3,
+                         fuzzy_strings=i ** 3))
+        resource_kwa += resources
+    project_locales = factories['project_locale'](batch_kwargs=pl_kwa)
+    resources = factories['resource'](batch_kwargs=resource_kwa)
+    [tr.update(dict(resource=resources[i % len(resources)]))
+     for i, tr
+     in enumerate(tr_kwa)]
```

Comment:
This is weird. Why not just make a for loop if you're not using the data structure you're creating?

Reply:
i got carried away with the functional programming - i can update